### PR TITLE
Add secrets to Density and Load tests

### PR DESCRIPTION
**Pull request by gmarek**
cc @jeremyeder @timstclair @sjug

### CODE REVIEW

**Comment by k8s-oncall** [comment]
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/37837)

File: test/e2e/density.go [hunk 1]
```diff
@@ -405,15 +416,28 @@ var _ = framework.KubeDescribe("Density", func() {
...
 			podThroughput := 20
 			timeout := time.Duration(totalPods/podThroughput)*time.Second + 3*time.Minute
 			// createClients is defined in load.go
 			clients, internalClients, err := createClients(numberOfCollections)
 			for i := 0; i < numberOfCollections; i++ {
-				name := fmt.Sprintf("density%v-%v-%v", totalPods, i, uuid)
 				nsName := namespaces[i].Name
+				secretNames := []string{}
+				for j := 0; j < itArg.secretsPerPod; j++ {
+					secretName := fmt.Sprintf("density-secret-%v-%v", i, j)
```
Comment (wojtek-t) [inline test/e2e/density.go]: I think we should configure it differently. In a real world, i don't think that all secrets in the system will be different. 
I think that what we should do is to:
- create a number of secrets (this number should be the parameters)
- mount (up to?) <secretsPerPod> random secrets to each pod

Having each pod a different set of secrets is a bit artificial.

File: test/utils/runners.go [hunk 2]
```diff
@@ -245,6 +248,12 @@ func (config *DeploymentConfig) create() error {
 		},
 	}
 
+	if len(config.SecretNames) > 0 {
+		volumes, mounts := getVolumesForSecrets(config.SecretNames)
```
Comment (wojtek-t) [inline test/utils/runners.go]: There are different ways of passing secrets. Volumes are one way. But there are other ways:
- as env vars:
https://github.com/kubernetes/kubernetes/blob/master/pkg/api/types.go#L1095
- image pull secrets:
https://github.com/kubernetes/kubernetes/blob/master/pkg/api/types.go#L1818
We should at least make it possible to use the all. 

So I would suggest changing ```getVolumesForSecrets ``` method to:
```attachSecrets(template *PodTemplateSpec, secretNames []string) ```
or sth like that
[even though in the first try you may want to mount all of them as volumes]

File: test/e2e/density.go (hunk 1 above)
Comment (gmarek) [inline test/e2e/density.go]: Discussed f2f - we mount the same 'secretsPerPod' number of secrets for all Pods in a single RC/RS/Dep.

File: test/utils/runners.go (hunk 2 above)
Comment (gmarek) [inline test/utils/runners.go]: Done. @wojtek-t PTAL

**Comment by gmarek** [comment]
@k8s-bot unit test this issue: #IGNORE
head was broken at the time

**Comment by gmarek** [comment]
@k8s-bot gke smoke test this issue: #IGNORE
head was broken at the time

**Comment by timstclair** [comment]
@timothysc

**Review by wojtek-t — COMMENTED** [review]
Just a single comments - other than that LGTM

File: test/utils/runners.go [hunk 3]
```diff
@@ -926,3 +941,61 @@ func NewSimpleWithControllerCreatePodStrategy(controllerName string) TestPodCrea
...
+
+func (config *SecretConfig) Stop() error {
+	if err := config.Client.Core().Secrets(config.Namespace).Delete(config.Name, &v1.DeleteOptions{}); err != nil {
+		return fmt.Errorf("Error deleting secret: %v", err)
+	}
+	config.LogFunc("Deleted secret %v/%v", config.Namespace, config.Name)
+	return nil
+}
+
+func attachSecrets(template *v1.PodTemplateSpec, secretNames []string) {
```
Comment (gmarek) [inline test/utils/runners.go]: Done. Applying LGTM

**Comment by wojtek-t** [comment]
lgtm - thanks!

**Pull request merged**